### PR TITLE
[CIS-658] Introduce trait collection reload stack

### PR DESCRIPTION
**Pull request by ManWithBear**
System calls `traitCollectionDidChange` in parent-first way, but our SDK relies on parent be able to override child properties.
To achieve it we wait store appearance reload closures into stack and on next runloop iteration call them in FILO order.

Known drawback of such approach:
- if parent's appearance reload calls `updateContent` on child, it will be performed twice (first by child's appearance reload, then by parent)

### CODE REVIEW

**Review by VojtaStavik — APPROVED** [review]
LGTM if it actually works :)

**Review by DominikBucher12 — APPROVED** [review]
nice

**Review by DominikBucher12 — APPROVED** [review]
Awesome

File: Sources/StreamChatUI/AppearanceSetting.swift [hunk 1]
```diff
@@ -58,6 +58,25 @@ private class _AppearanceStorage {
...
+    ///   1. setUp()
+    ///   2. setUpLayout()
+    ///   3. defaultAppearance()
+    ///   4. 👉 <custom rules>
+    ///   5. setUpAppearance()
+    ///   6. updateContent()
+    /// ```
+    ///
+    /// - Important: The closure can be executed multiple times for the same instance or not executed at all. All changes
+    /// done in the closure should be idenpotent.
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/AppearanceSetting.swift:75]:
```suggestion
    /// done in the closure should be idempotent.
```

**Review by nuno-vieira — APPROVED** [review]
Small typo, other then that LGTM 👍

**Pull request merged**